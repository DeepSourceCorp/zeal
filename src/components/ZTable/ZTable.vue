<template>
  <table class="z-table p-8 border border-ink-200 border-collapse w-full">
    <thead class="w-full">
      <slot name="head"></slot>
    </thead>
    <tbody class="w-full">
      <slot name="body"></slot>
    </tbody>
  </table>
</template>

<script>
export default {
  name: 'ZTable',
  props: {
    height: {
      default: '',
      type: String
    }
  },
  computed: {},
  methods: {}
}
</script>
