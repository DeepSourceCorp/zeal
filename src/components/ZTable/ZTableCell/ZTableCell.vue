<template>
  <td class="z-table-cell py-2.5 px-5 flex-1">
    <slot></slot>
  </td>
</template>

<script>
export default {
  name: 'ZTableCell'
}
</script>
