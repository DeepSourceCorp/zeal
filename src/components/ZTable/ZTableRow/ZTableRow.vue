<template>
  <tr class="z-table-row border-b border-slate-400 flex w-full">
    <slot></slot>
  </tr>
</template>

<script>
export default {
  name: 'ZTableRow'
}
</script>
