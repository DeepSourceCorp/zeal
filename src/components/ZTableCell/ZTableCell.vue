<template>
  <td class="z-table-cell py-2.5 px-5 flex-1" :class="[`text-${textAlign}`]">
    <slot></slot>
  </td>
</template>

<script>
export default {
  name: 'ZTableCell',
  props: {
    textAlign: {
      default: 'left',
      type: String
    }
  },
  computed: {},
  methods: {}
}
</script>
