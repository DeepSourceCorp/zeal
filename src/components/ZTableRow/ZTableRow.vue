<template>
  <tr class="z-table-row border-b border-ink-300 flex w-full">
    <slot></slot>
  </tr>
</template>

<script>
export default {
  name: 'ZTableRow',
  props: {
    height: {
      default: '',
      type: String
    }
  },
  computed: {},
  methods: {}
}
</script>
