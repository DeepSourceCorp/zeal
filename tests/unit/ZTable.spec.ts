import ZTable from '../../src/components/ZTable'
import ZTableRow from '../../src/components/ZTableRow'
import ZTableCell from '../../src/components/ZTableCell'

import { mount } from '@vue/test-utils'

const Table = {
  template: `<z-table>
        <template v-slot:head>
          <z-table-row>
            <z-table-cell>Heading 1</z-table-cell>
            <z-table-cell>Heading 2</z-table-cell>
          </z-table-row>
        </template>
        <template v-slot:body>
          <z-table-row>
            <z-table-cell>Content Row 1</z-table-cell>
            <z-table-cell>Content Row 1</z-table-cell>
          </z-table-row>
          <z-table-row>
            <z-table-cell>Content Row 2</z-table-cell>
            <z-table-cell>Content Row 2</z-table-cell>
          </z-table-row>
        </template>
      </z-table>`,
  components: {
    ZTable,
    ZTableRow,
    ZTableCell
  }
}

describe('Table', () => {
  it('renders a default table', () => {
    const wrapper = mount(Table)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
