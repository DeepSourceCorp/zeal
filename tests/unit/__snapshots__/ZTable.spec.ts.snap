// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders a default table 1`] = `
<table class="z-table p-8 border border-ink-200 border-collapse w-full">
  <thead class="w-full">
    <tr class="z-table-row border-b border-ink-300 flex w-full">
      <td class="z-table-cell py-2.5 px-5 flex-1">Heading 1</td>
      <td class="z-table-cell py-2.5 px-5 flex-1">Heading 2</td>
    </tr>
  </thead>
  <tbody class="w-full">
    <tr class="z-table-row border-b border-ink-300 flex w-full">
      <td class="z-table-cell py-2.5 px-5 flex-1">Content Row 1</td>
      <td class="z-table-cell py-2.5 px-5 flex-1">Content Row 1</td>
    </tr>
    <tr class="z-table-row border-b border-ink-300 flex w-full">
      <td class="z-table-cell py-2.5 px-5 flex-1">Content Row 2</td>
      <td class="z-table-cell py-2.5 px-5 flex-1">Content Row 2</td>
    </tr>
  </tbody>
</table>
`;
